packages/next/client/image.tsx

```diff
@@ -444,9 +468,19 @@ export default function Image({
         `Image with src "${src}" has both "priority" and "loading='lazy'" properties. Only one should be used.`
       )
     }
-    if (sizes && layout !== 'fill' && layout !== 'responsive') {
+    if (layout === 'raw' && (objectFit || objectPosition)) {
+      throw new Error(
+        `Image with src "${src}" has "layout='raw'" and 'objectFit' or 'objectPosition'. For raw images, these and other styles should be specified using the 'style' attribute.`
+      )
+    }
```

Comment:
An earlier version of this PR did exactly what you're suggesting (handling objectFit and objectPosition when passed as a prop), but after some discussion with @styfle I ended up changing it to the current method. I think either way it's not ideal, but this way is at least consistent in that when in raw mode all styles are applied via the `style` prop. Having some styles automatically translated from props to styles seems like it could create the expectation that other styles could be passed the same way.

Also, this implementation simplifies the code a bit.

Reply:
I understand and I tend to agree with you. I'm just pointing out this will create some friction for users because it's easier to code. It might not be a huge deal, I'll give you that :)

Personally, I think it was a mistake to expose CSS properties as props. As you mentioned, it creates the expectation that other styles could be passed the same way. It's preferable to separate concerns and `objectFit`/`objectPosition` are definitely CSS concerns, not JS concerns.

Reply:
The reason those props were added is because you can't use inline style:

```jsx
<Image src={src} layout="fill" style={{objectFit: "cover"}} />
```

So the workaround was adding these helper props to assist:

```jsx
<Image src={src} layout="fill" objectFit="cover" />
```

I think we can throw an error when `layout=raw` and `objectFit`/`objectPosition` are used just to make it clear that `style` should be used instead.

Reply:
> The reason those props were added is because you used to not be able to use inline style:
> 
> ```js
> <Image src={src} layout="fill" style={{objectFit: "cover"}} />
> ```

I would add it was also added because we can't target the images easily by pure CSS selectors. Which is still a much needed feature one year after the launch of `next/image` :(

I would argue it was also a bit arbitrary that only `objectFit` and `objectPosition` were targeted to be props in this context. A class on each wrapper would have made it super easy to add all the CSS styling you want without resorting to using props, inline styles or DOM bloating. Separation of concerns is a big thing for a lot of people.